SDL/MacOSBarItems.mm

```diff
@@ -0,0 +1,326 @@
+            // set the new item to be the selected one
+            dispatch_async(dispatch_get_main_queue(), ^{
+                for (NSMenuItem *item in theMenu.itemArray)
+                    item.state = [self controlStateForBool:g_Config.sAudioDevice == audioDeviceList[item.tag]];
+            });
+        }
+    }];
+    
+    return theMenu;
+}
+
+-(void) setAudioItem: (NSMenuItem *)sender {
+    std::vector<std::string> audioDeviceList;
+    SplitString(System_GetProperty(SYSPROP_AUDIO_DEVICE_LIST), '\0', audioDeviceList);
+    
+    std::string theItemSelected = audioDeviceList[sender.tag];
+    if (theItemSelected == g_Config.sAudioDevice)
+        return; // device already selected
+    
+    g_Config.sAudioDevice = theItemSelected;
+    for (NSMenuItem *item in sender.menu.itemArray) {
+        item.state = [self controlStateForBool:g_Config.sAudioDevice == theItemSelected];
+    }
+    
+    System_SendMessage("audio_resetDevice", "");
+}
+
+#define TOGGLE_METHOD(name, ConfigValueName, ...) \
+-(void)toggle##name: (NSMenuItem *)item { \
+ConfigValueName = !ConfigValueName; \
+__VA_ARGS__; /* for any additional updates */ \
+item.state = [self controlStateForBool: ConfigValueName]; \
+}
+
+TOGGLE_METHOD(Sound, g_Config.bEnableSound)
+TOGGLE_METHOD(AutoFrameSkip, g_Config.bAutoFrameSkip, g_Config.updateAfterSettingAutoFrameSkip())
+TOGGLE_METHOD(SoftwareRendering, g_Config.bSoftwareRendering)
+TOGGLE_METHOD(FullScreen, g_Config.bFullScreen, System_SendMessage("toggle_fullscreen", g_Config.UseFullScreen() ? "1" : "0"))
+TOGGLE_METHOD(VSync, g_Config.bVSync)
+#undef TOGGLE_METHOD
+
+-(void)setToggleShowCounterItem: (NSMenuItem *)item {
+    [self addOrRemoveInteger:(int)item.tag to:&g_Config.iShowStatusFlags];
+    item.state = [self controlStateForBool:g_Config.iShowStatusFlags & item.tag];
+}
+
+-(void)addOrRemoveInteger: (int)integer to: (int *)r {
+    if (integer & *r) {
+        *r -= integer;
+    } else {
+        *r |= integer;
+    }
+}
+
+-(void)setCurrentGPUBackend: (NSMenuItem *)sender {
+    std::vector<GPUBackend> allowed = [self allowedGPUBackends];
+    if (allowed.size() == 1) {
+        printf("only one item, bailing");
+        return;
+    }
+    
+    g_Config.iGPUBackend = (int)(allowed[sender.tag]);
+    sender.state = NSControlStateValueOn;
+    
+    for (NSMenuItem *item in sender.menu.itemArray) {
+        // deselect the previously selected item
+        if (item.state == NSControlStateValueOn && item.tag != sender.tag) {
+            item.state = NSControlStateValueOff;
+            break;
+        }
+    }
+}
+
+-(NSControlStateValue) controlStateForBool: (BOOL)boolValue {
+    return boolValue ? NSControlStateValueOn : NSControlStateValueOff;
+}
+
+-(std::vector<GPUBackend>)allowedGPUBackends {
+    std::vector<GPUBackend> allBackends = {
+        GPUBackend::OPENGL, GPUBackend::VULKAN,
+        GPUBackend::DIRECT3D11, GPUBackend::DIRECT3D9
```

Comment:
I'll be impressed if anyone ever gets these two working on Cocoa.

-[Unknown]

Reply:
Oh yeah that's basically impossible, heh, I did think 'hey these can't ever really work on macOS or be enabled' but then got distracted by something else, will remove :p